llvm/lib/IR/AutoUpgrade.cpp

```diff
@@ -5155,7 +5155,39 @@ struct StrictFPUpgradeVisitor : public InstVisitor<StrictFPUpgradeVisitor> {
 };
 } // namespace
 
-void llvm::UpgradeFunctionAttributes(Function &F) {
+static void
+CopyModuleAttributeToFunction(Function &F, StringRef FnAttrName,
+                              StringRef ModAttrName,
+                              std::pair<StringRef, StringRef> Values) {
```

Comment:
Can you add a comment about what the pair of StringRef are supposed to refer to? The parameter identifier of Values doesn't betray what the caller should be passing here.